### ISSUE TITLE
Support "follow-referral" mode

### ISSUE THREAD

**Issue by zabhi** [vesse/node-ldapauth-fork#88]
I am using this module via passport-ldap. The directory I am working with is setup with referrals to a second LDAP server.
The library is unable to chase the referrals to the target server.

Is this the expected? Can I do something to make it follow the referrals?

**Comment by vesse** [comment]
Yes, expected in a sense that it is not implemented. The underlying `ldapjs` library handles all LDAP related things really, and it does not follow referrals automatically. Quick search revealed that it should return the referrals so implementing chasing should be possible. `node-activedirectory` has this implemented in [this commit](https://github.com/gheeres/node-activedirectory/commit/a06fbada2153fe555fb21f6063465050f98d8ae4) which should give some insight how it could be implemented. 

Unfortunately I find it unlikely that I'd have time to do this any time soon, especially when I don't have access to a setup like yours. PR would of course be very welcome.
